### PR TITLE
Hidden stages

### PR DESCRIPTION
**Pull request by bcjordan**
This is @Bjvanminnen's PR formerly known as https://github.com/code-dot-org/code-dot-org/pull/10944

Accidentally merged in via Circle's new merge button, reverted in code-dot-org/code-dot-org#10949, this is a revert of that for @Bjvanminnen as an option to continue working from.

### CODE REVIEW

**Comment by Bjvanminnen** [comment]
@islemaster Feel free to comment either on this PR or the original (where I already have a couple of line comments)

File: apps/src/code-studio/hiddenStageRedux.js [hunk 1]
```diff
@@ -81,7 +88,7 @@ export function allowHideable() {
 
 
 export function getHiddenStages(scriptName) {
-  return (dispatch, getState) => {
+  return (dispatch) => {
```
Comment (islemaster) [inline apps/src/code-studio/hiddenStageRedux.js]: Nit: Unneeded parens.

File: apps/src/code-studio/hiddenStageRedux.js [hunk 2]
```diff
@@ -91,16 +98,38 @@ export function getHiddenStages(scriptName) {
...
       console.error(err);
     });
   };
 }
+
+// utils
+
+/**
+ * Helper to determine whether a stage is hidden for a given section. If no
+ * section is given,
```
Comment (islemaster) [inline apps/src/code-studio/hiddenStageRedux.js]: Incomplete comment,

File: apps/src/code-studio/hiddenStageRedux.js [hunk 3]
```diff
@@ -91,16 +98,38 @@ export function getHiddenStages(scriptName) {
...
+ */
+export function isHiddenFromState(bySection, sectionId, stageId) {
+  // if we don't have a sectionId, we must be a student
+  if (sectionId === null ){
```
Comment (islemaster) [inline apps/src/code-studio/hiddenStageRedux.js]: Nit: Trailing space on expression?

**Pull request merged**